jnsgruk/zinc-k8s-operator
# [JUJU-2032] Integration tests can be run against zinc from charmhub

Add an optional channel param to the pytest suite which, if provided, deploys zinc from this channel on CharmHub instead of building from source

The motivation behind this is that building from source can take a very long time, and isn't necessary in all contexts.

Default behaviour is maintained is the option is absent

### QA Steps

Verify integration tests run as expected:

Builds charm from source
```
tox -e integration
```

Deploys charm from charmhub
```
tox -e integration -- --channel latest/stable
```

(Note: There are log messages which indicate which is being performed)
